Issue 1829: Upgrade to latest version of Nuget

j4ckofalltrades opened the pull request:
Upgrade to Nuget v5.7.0.

# Pull Request Description

This pull request is a sub-task of strongbox/strongbox#1829

# Acceptance Test

* [x] Running `./build.sh` successfully builds all images.

# Questions

* Does this pull request somehow break backward compatibility? 
  * [ ] Yes, it will require fixing build jobs.
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [x] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

j4ckofalltrades committed:
Issue 1829: Upgrade to latest version of Nuget

Upgrade to Nuget v5.7.0.

j4ckofalltrades committed:
Issue-1829: Revert choco to use Nuget v3.4

Reverted choco docker images to use Nuget v3.4, as this change should be
done after the Nuget docker images have been published to Docker hub
(changes to the choco docker images were done in error).

steve-todorov commented on images/alpine/jdk11/Dockerfile.alpine.jdk11-mvn3.6-mono5-nuget5.7:
```diff
@@ -2,13 +2,13 @@ ARG SNAPSHOT=""
-ENV NUGET_V3_EXEC="mono ${NUGET_PATH}"
+ENV NUGET_V5_VERSION=$NUGET_VERSION
+ENV NUGET_V5_PATH=$NUGET_PATH
+ENV NUGET_V5_EXEC="mono ${NUGET_PATH}"
```
Renaming the file is technically correct, but in this particular case you should revert the `nuget3.4` file naming and add 2 new files for `jdk8-nuget5.7 ` and `jdk11-nuget5.7`. 

Also, you would need to add `Dockerfile.alpine.jdk8-mvn3.6-mono5-nuget5.7-choco0.10` and `Dockerfile.alpine.jdk11-mvn3.6-mono5-nuget5.7-choco0.10` files which extend from the `jdk8-nuget5.7` and `jdk11-nuget5.7` images.

You **MUST** keep the `ENV NUGET_V3_*` env vars named as they are, because this is used in the [strongbox-web-integration-tests](https://github.com/strongbox/strongbox-web-integration-tests) project. I've added a new task for this - strongbox/strongbox#1905. If you are interested in fixing that as well - comment under the issue.

j4ckofalltrades replied:
Got it. I've applied needed changes.

steve-todorov reviewed (changes requested):
Hi @j4ckofalltrades, 

Thank you for your contribution!

I've left a few notes on your changes. 

Thanks!

j4ckofalltrades committed:
Issue-1829: Revert Nuget dockerfile version to 3.4

The strongbox-web-integration-tests expects the NUGET_V3 env variables
to be set, so the Nuget upgrade will be done in a separate dockerfile.

j4ckofalltrades committed:
Issue-1829: Create Dockerfile for latest Nuget version

Setup docker image with Nuget version v5.7.0.
Nuget env variables have also been renamed which will be used in
issue #1905.

Refs #1829

j4ckofalltrades commented on images/alpine/jdk11/Dockerfile.alpine.jdk11-mvn3.6-mono5-nuget5.7:
```diff
@@ -0,0 +1,24 @@
+# setup-nuget.sh
+ENV NUGET_VERSION=5.7.0
+ENV NUGET_PATH=/usr/local/bin/nuget-${NUGET_VERSION}.exe
+
```
@steve-todorov I've renamed env variables here which will be used in strongbox/strongbox#1905.

steve-todorov replied:
This is a **separate** task and needs to be done as part of a separate PR. Please revert it to the original ENV vars. :)

j4ckofalltrades replied:
Oops... Okay will revert change and rename env vars.

j4ckofalltrades committed:
Issue-1829: Revert NUGET env vars naming

Refs #1829

steve-todorov reviewed (approved):
LGTM! Thanks for your contribution! :)
